src/main/java/com/paneedah/weaponlib/EntityProjectile.java

```diff
@@ -243,7 +244,9 @@ public void onUpdate() {
              if(rtr != null) {
                  IBlockState state = compatibility.world(this).getBlockState(rtr.getBlockPos());
                  if(state.getMaterial() == Material.GLASS) {
-                     this.world.destroyBlock(rtr.getBlockPos(), true);
+                     if(!penetratesGlass)
+                         this.world.destroyBlock(rtr.getBlockPos(), true);
```

Comment:
It should also check for `bulletBreakGlass` before actually breaking it

Reply:
Do you think this is reasonable? When glass penetration is opened, bullets will penetrate the glass regardless of whether it can be broken or not

Reply:
There is a need, since someone may want it to penetrate but not actually break it sometimes.

Reply:
That sounds amazing. The ability to penetrate glass without breaking it. 

Would be really helpful for servers who still want glass to be dangerous but to prevent grief.

Reply:
> Do you think this is reasonable? When glass penetration is opened, bullets will penetrate the glass regardless of whether it can be broken or not

You should do something like `(!penetratesGlass && bulletBreakGlass)`

Reply:
Will present two effects (leaving/not leaving particles when bullets penetrate)

Reply:
Now when bullets penetrate, they must leave particles behind